src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs

```diff
@@ -176,22 +177,32 @@ public WebSocketsTransport(WebSocketOptions options, Channel<byte[]> application
                     {
                         try
                         {
-                            _logger.SendPayload(_connectionId, buffer.Length);
+                            _logger.SendPayload(_connection.ConnectionId, buffer.Length);
+
+                            var webSocketMessageType = _options.WebSocketMessageType;
+                            var transferModeFeature = _connection.Features.Get<ITransferModeFeature>();
+                            if (transferModeFeature != null)
```

Comment:
Just pass this directly into the ctor of the WebSocketTransport instead of getting the feature each time. It doesn't change dynamically (or shouldn't).